libraries/esp8266/examples/MMU48K/MMU48K.ino

```diff
@@ -0,0 +1,354 @@
+      out.printf_P(PSTR("Load/Store exception by reading short from iRAM"));
+      out.println();
+      out.flush();
+      out.printf_P(PSTR("Read short from iRAM, 0x%04X at %p"), probe_s[0], probe_s);
+      out.println();
+      break;
+    case 'S': {
+        out.printf_P(PSTR("Load/Store exception by writing short to iRAM"));
+        out.println();
+        short int val = 0x0AA0;
+        out.printf_P(PSTR("Write short, 0x%04X, to iRAM at %p"), val, probe_s);
+        out.println();
+        out.flush();
+        probe_s[0] = val;
+        out.printf_P(PSTR("Read short back from iRAM, 0x%04X at %p"), probe_s[0], probe_s);
+        out.println();
+        break;
+      }
+    case 'R':
+      out.printf_P(PSTR("Restart, ESP.restart(); ..."));
+      out.println();
+      ESP.restart();
+      break;
+    case 'p':
+      out.println(F("Time to panic()!"));
+      panic();
+      break;
+    case '0':
+      out.println(F("Crashing by dividing by zero."));
+      out.printf_P(PSTR("This should not print %d"), divideA_B(1, 0));
+      out.println();
+      break;
+    case '\r':
+      out.println();
+    case '\n':
+      break;
+    case '?':
+      out.println();
+      out.println(F("Press a key + <enter>"));
+      out.println(F("  R    - Restart, ESP.restart();"));
+      out.println(F("  t    - exception vs inline method timing info."));
+      out.println(F("  ?    - Print Help"));
+      out.println();
+#if defined(NON32XFER_HANDLER)
+      out.println(F("Test exception handling with non-32 bit transfer handler:"));
+#else
+      out.println(F("Crash with:"));
+#endif
+      out.println(F("  b    - read byte, Load/Store exception"));
+      out.println(F("  B    - write byte, Load/Store exception"));
+      out.println(F("  s    - read short, Load/Store exception"));
+      out.println(F("  S    - write short, Load/Store exception"));
+#if defined(NON32XFER_HANDLER)
+      out.println();
+      out.println(F("Crash with:"));
+#endif
+      out.println(F("  c    - read byte, Load/Store exception outside of handler range"));
+      out.println(F("  9    - read short, Unaligned exception"));
+
+      out.println(F("  0    - Divide by zero, exception(0);"));
+      out.println(F("  p    - panic();"));
+      out.println();
+      break;
+    default:
+      out.printf_P(PSTR("\"%c\" - Not an option?  / ? - help"), hotKey);
+      out.println();
+      break;
+  }
+}
+
+
+void serialClientLoop(void) {
+  if (Serial.available() > 0) {
+    int hotKey = Serial.read();
+    processKey(Serial, hotKey);
+  }
+}
+
+void loop() {
+  serialClientLoop();
+}
+
+
+int __attribute__((noinline)) divideA_B(int a, int b) {
+  return (a / b);
+}
+
+#if 0
+#ifdef MMU_SEC_HEAP
+extern "C" void _text_end(void);
+
+extern "C" void umm_init_iram(void) {
+  uint32_t sec_heap = (uint32_t)_text_end + 32;
+  sec_heap &= ~7;
+  size_t   sec_heap_sz = 0xC000UL - (sec_heap - 0x40100000UL);
+
+  umm_init_iram_ex((void *)sec_heap, sec_heap_sz, true);
+}
+#endif
```

Comment:
Cleanup?

Reply:
I'll remove this for now. 

Someday I'll create an example to illustrate how to reserve some of the IRAM from the heap in order to create an always available IRAM buffer that can be used across boots. Examples: a crash stack save buffer, a diagnostic message print buffer, etc.

This block shows that there is some external control over defining the IRAM heap; however, it needs comments and an example.

Reply:
Done. Added example IramReserve.ino.